umbraco/Umbraco-CMS
# Properly set / unset Culture flag when "vary by Culture" is changed

### Prerequisites

- [X ] I have added steps to test this contribution in the description below

Fixes #6691 

### Description
Toggling the "Allow varying by culture" setting on a content type or property type now only touches the `ContentVariation.Culture` flag, rather than directly setting either `ContentVariation.Nothing` or `ContentVariation.Culture`. This means the `ContentVariation.Segment` flag is always cleared which is not correct.

### How to test
This fix should not break any existing functionality (it will still switch between `Nothing` and `Culture` until segments are available in Umbraco), so the test would be to run whatever existing tests are available to test the "Allow varying by culture" behavior. It should be the same.

---
_This item has been added to our backlog [AB#3855](https://umbraco.visualstudio.com/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/3855)_
